src/canvas.class.js

```diff
@@ -64,7 +64,56 @@
       this.requestRenderAllBound = this.requestRenderAll.bind(this);
       this._initStatic(el, options);
       this._initInteractive();
-      this._createCacheCanvas();
+      this._createCacheCanvas(); 
+
+      this._initHighlightDrawing();
+    },
+
+    // In-memory canvas element to support highlight drawing
+    highlightDrawingCanvasEl: null,
+
+    _initHighlightDrawing: function() {
+      console.log('fabric.Canvas::_initHighlightDrawing');
+      
+      //(1) Init off-screen buffer
+      if (!this.highlightDrawingCanvasEl) {
+        var canvasBuffer = document.createElement('canvas');
+        canvasBuffer.width  = this.width;
+        canvasBuffer.height = this.height;
+        this.highlightDrawingCanvasEl = canvasBuffer;
+      }
+
+      //(2) Insert Image representing buffer onto the stage
+      if (this.getHighlightLayerIndex() < 0) {
+        var fabricImageBuffer = new fabric.Image(this.highlightDrawingCanvasEl);
+        fabricImageBuffer.selectable = false;
+        fabricImageBuffer.isHighlightLayer = true;
+        //fabricImageBuffer.imageSmoothing = false;
+        this.add(fabricImageBuffer);
```

Comment:
I believe this will result in an entry to the undo/redo history. Options to avoid that are to clear the history after initialization, or to suppress the history before the .add()

Reply:
Tried to make this method in a way we could reuse for deserialization. How is the undo/redo suppressed there? This might help us answer where it goes, but both of these options would work perfectly fine :)

Reply:
Currently the undo/redo history is NOT suppressed in deserialization. This is good because it means that as the items are loaded in they are added to the history - so you can still undo things after loading from a browser refresh. But I think we DO wanna suppress this one so that the user can't press 'undo' as a first action and have this get removed.

Reply:
Got it. I think then we should change it so on deserialization it does clear the history, at least for now until we potentially fix the undo-from-saved-json functionality.